Use common strings in emulated roku

GauthamVarmaK opened the pull request:
## Proposed change
Use common strings in emulated_roku integration as described in #40578

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #40578
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

GauthamVarmaK committed:
Use common strings in emulated roku

Replace IP and port with references from common strings

GauthamVarmaK commented:
> could 
> https://github.com/home-assistant/core/blob/c1aa10f63982520556aad8d70484bca32c30c24e/homeassistant/components/emulated_roku/strings.json#L4
> be
> ```python
>  "abort": { "already_configured": "[%key:common::config_flow::abort::already_configured_device%]" }, 
> 
> ```
> becouse
> https://github.com/home-assistant/core/blob/c1aa10f63982520556aad8d70484bca32c30c24e/homeassistant/components/emulated_roku/config_flow.py#L33-L34

Was thinking the same, but not sure if it was supposed to mean the same, if anyone can confirm will add the commit.

MartinHjelmare commented:
Yeah, that's a good replacement.

MartinHjelmare commented on homeassistant/components/emulated_roku/strings.json:
```diff
@@ -5,10 +5,10 @@
-          "advertise_port": "Advertise port",
-          "host_ip": "Host IP",
-          "listen_port": "Listen port",
+          "advertise_ip": "Advertise [%key:common::config_flow::data::ip%]",
```
We try to avoid mixing in references in a sentence.

GauthamVarmaK replied:
👍 Reverted

GauthamVarmaK replied:
👍 Removed references, although edited the values directly to bring consistency (Ex: IP-->IP Address)

GauthamVarmaK committed:
Update strings.json

*Replace References in mix with sentence with their values.
*Replace name_exists with already_configured

GauthamVarmaK committed:
Replace references in config_flow

MartinHjelmare reviewed (approved):
Thanks!
